CRM/Core/Payment.php

```diff
@@ -1277,9 +1277,24 @@ public function doPayment(&$params, $component = 'contribute') {
    *
    * @param array $params
    *
+   * @return array
    * @throws \Civi\Payment\Exception\PaymentProcessorException
    */
-  public function doRefund(&$params) {}
+  public function doRefund($params) {
+    // Mandatory Params:
+    // trxn_id: The transaction ID that the payment processor uses to identify the payment to refund.
+    // amount: The amount to refund (eg. 12.05). This should always be specified in "dot" notation with no currency symbol.
+    return [
+      // refund_status_id would normally return the contribution_status_id Completed or Failed
+      'refund_status_id' => CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Completed'),
+      // The refund trxn_id may be different from the trxn_id of the original payment.
+      // If it is the same then trxn_id should be copied to refund_trxn_id
+      'refund_trxn_id' => NULL,
```

Comment:
OK - that makes sense - you have changed it from altering params to having defined return vars here too - I'm OK with switching over